src/components/dialogs/voltage-level/modification/voltage-level-modification-dialog.js

```diff
@@ -0,0 +1,195 @@
+import VoltageLevelModificationForm from './voltage-level-modification-form';
+import {
+    EQUIPMENT_ID,
+    EQUIPMENT_NAME,
+    HIGH_SHORT_CIRCUIT_CURRENT_LIMIT,
+    HIGH_VOLTAGE_LIMIT,
+    LOW_SHORT_CIRCUIT_CURRENT_LIMIT,
+    LOW_VOLTAGE_LIMIT,
+    NOMINAL_VOLTAGE,
+    SUBSTATION_ID,
+} from '../../../utils/field-constants';
+import yup from '../../../utils/yup-config';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import {
+    FetchStatus,
+    fetchVoltageLevel,
+    modifyVoltageLevel,
+} from '../../../../utils/rest-api';
+import { useOpenShortWaitFetching } from '../../commons/handle-modification-form';
+import { FORM_LOADING_DELAY } from '../../../network/constants';
+
+const emptyFormData = {
+    [EQUIPMENT_ID]: '',
+    [EQUIPMENT_NAME]: '',
+    [SUBSTATION_ID]: null,
+    [NOMINAL_VOLTAGE]: null,
+    [LOW_VOLTAGE_LIMIT]: null,
+    [HIGH_VOLTAGE_LIMIT]: null,
+    [LOW_SHORT_CIRCUIT_CURRENT_LIMIT]: null,
+    [HIGH_SHORT_CIRCUIT_CURRENT_LIMIT]: null,
+};
+
+const schema = yup.object().shape({
+    [EQUIPMENT_ID]: yup.string().required(),
+    [EQUIPMENT_NAME]: yup.string().nullable(),
+    [SUBSTATION_ID]: yup.string().nullable(),
+    [NOMINAL_VOLTAGE]: yup.number().nullable(),
+    [LOW_VOLTAGE_LIMIT]: yup.number().nullable(),
+    [HIGH_VOLTAGE_LIMIT]: yup.number().nullable(),
+    [LOW_SHORT_CIRCUIT_CURRENT_LIMIT]: yup.number().nullable(),
+    [HIGH_SHORT_CIRCUIT_CURRENT_LIMIT]: yup.number().nullable(),
+});
+
+const VoltageLevelModificationDialog = ({
+    editData,
+    currentNode,
+    studyUuid,
+    isUpdate,
+    editDataFetchStatus,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode?.id;
+    const { snackError } = useSnackMessage();
+    const [voltageLevelInfos, setVoltageLevelInfos] = useState(null);
+    const [dataFetchStatus, setDataFetchStatus] = useState(FetchStatus.IDLE);
+
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const { reset, setValue } = methods;
+
+    useEffect(() => {
+        if (editData) {
+            reset(
+                {
+                    [EQUIPMENT_ID]: editData?.equipmentId ?? '',
+                    [EQUIPMENT_NAME]: editData?.equipmentName?.value ?? '',
+                    [SUBSTATION_ID]: editData?.substationId?.value ?? null,
+                    [NOMINAL_VOLTAGE]: editData?.nominalVoltage?.value ?? null,
+                    [LOW_VOLTAGE_LIMIT]:
+                        editData?.lowVoltageLimit?.value ?? null,
+                    [HIGH_VOLTAGE_LIMIT]:
+                        editData?.highVoltageLimit?.value ?? null,
+                    [LOW_SHORT_CIRCUIT_CURRENT_LIMIT]:
+                        editData?.ipMin?.value ?? null,
+                    [HIGH_SHORT_CIRCUIT_CURRENT_LIMIT]:
+                        editData?.ipMax?.value ?? null,
```

Comment:
you can use editData only here without '?' because you have already did the check